operator/pkg/helmreconciler/prune.go

```diff
@@ -105,6 +105,10 @@ func (h *HelmReconciler) Prune(manifests name.ManifestMap, all bool) error {
 // during reconciliation process of controller.
 // It returns the install status and any error encountered.
 func (h *HelmReconciler) PruneControlPlaneByRevisionWithController(iopSpec *v1alpha1.IstioOperatorSpec) (*v1alpha1.InstallStatus, error) {
+	// flushes all object caches when exec prune
+	scope.Info("Flush object caches")
+	cache.FlushObjectCaches()
```

Comment:
flushing the object cache will clean the cache for all objects when we only want to prune specific revision control plane resources.